### PR TITLE
docs:docusaurus-new-doc

### PR DESCRIPTION
**Pull request by annehubtype**
_Remember to tag the PR with **WIP** tag if it's not ready to be merged_.  
[Reference](https://blog.codeminer42.com/on-writing-a-great-pull-request-37c60ce6f31d/)

## Description

Please review the new format of the documentation with Docusaurus - still work in progress:
The content does not need to be reviewed for now, **just the visual output**. Thanks!
--> See **doc-botonic/website/README.md** for more information.
 
If you agree with this version, we will proceed in the deletion of the older docZ repository.

## Testing

The pull request...

- [ ] has unit tests
- [ ] has integration tests
- [ ] doesn't need tests because... **[provide a description]**

### CODE REVIEW

File: doc-botonic/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM node:lts
```
Comment (dpinol) [inline doc-botonic/Dockerfile:1]: docker forever! :heart:

**Comment by dpinol** [comment]
See 1st line of PR template (select WIP label):
"Remember to tag the PR with WIP tag if it's not ready to be merged."

**Comment by dpinol** [comment]
Good work!

Is there a REAME.md on the top folder with instructions or link to docusaurus info on how to generate the docs?

**Comment by marcpdw** [comment]
Any recommmendation to review this? I guess we wouldn't need to read the whole documentation, right?

**Comment by annehubtype** [comment]
> Any recommmendation to review this? I guess we wouldn't need to read the whole documentation, right?

I added more info in the PR, sorry : "The content does not need to be reviewed for now, just the visual output. Thanks!" ;)

**Comment by annehubtype** [comment]
> Good work!
> 
> Is there a REAME.md on the top folder with instructions or link to docusaurus info on how to generate the docs?

Yes here: doc-botonic/website/README.md 
I've added the path in the PR. Thanks!

File: doc-botonic/docs/faq.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Hubtype offers the best developer experience when it comes to deploying and integrating your Botonic bot.
+</details>
+
+><details><summary><b>How do I integrate a chat in a website?</b></summary>
+Once you deploy a bot,  you get a link to integrate different channels, including the "webchat" channel, which is a JavaScript snippet that you can embed in your website.
+</details>
+
+><details><summary><b>Can I edit my chat  styles?</b></summary>
+This action is not possible except for webviews as some of these apps (like Facebook Messenger) allow you to embed html pages inside their chat window.
+Hubtype provides support for web chat (JS SDK) and in-app chat (iOS/Android SDK). In these cases, you can customize the look and feel of the chat window.
```
Comment (ericmarcos) [inline doc-botonic/docs/faq.md:18]: I would rephrase this by saying that "yes, you can ..." and at the end adding the exception "however, you can't change the style of messaging apps like Whatsapp... in these cases you can only customise webviews..."

File: doc-botonic/docs/faq.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+</details>
+
+><details><summary><b>How do I delete a bot?</b></summary>
+Go to https://app.hubtype.com/bots/all and select your bot. Then click on the **Settings** tab and on **Delete this bot**.
+</details>
+
+><details><summary><b>How to deal with NLP/NLU?</b></summary>
+You must integrate Dialogflow or Watson. The integration allows you to use regular expressions to route intents to actions. Check out the NLU example in our github repo.
+</details>
```
Comment (ericmarcos) [inline doc-botonic/docs/faq.md:27]: This has changed. Now we should encourage the use of our NLU plugin and then mention that you can integrate Dialogflow/Watson/Luis/Inbenta too.

File: doc-botonic/docs/faq.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+><details><summary><b>Where can I deploy my Botonic bots?</b></summary>
+The easiest way is to deploy it to hubtype.com cloud. Hubtype sets up all the infrastructure you need, scales automatically and manages third-party integrations.
+You can also to run it and deploy anywhere. You just need some developement to make the integration with messaging channels work.
+</details>
+
+><details><summary><b>How do I connect Botonic with Zendesk?</b></summary>
+Although Zendesk integration is not a Botonic open-source plugin like DialogFlow, you can do it by deploying your bot and contact the Botonic team on Slack. They will help you with the next steps.
```
Comment (ericmarcos) [inline doc-botonic/docs/faq.md:35]: This is very specific, I'm not sure if it makes sense in the FAQ.

File: doc-botonic/docs/faq.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+Go to https://app.hubtype.com/bots/all and select your bot. Then click on the **Settings** tab and on **Delete this bot**.
+</details>
+
+><details><summary><b>How to deal with NLP/NLU?</b></summary>
+You must integrate Dialogflow or Watson. The integration allows you to use regular expressions to route intents to actions. Check out the NLU example in our github repo.
+</details>
+
+><details><summary><b>Where can I deploy my Botonic bots?</b></summary>
+The easiest way is to deploy it to hubtype.com cloud. Hubtype sets up all the infrastructure you need, scales automatically and manages third-party integrations.
+You can also to run it and deploy anywhere. You just need some developement to make the integration with messaging channels work.
```
Comment (ericmarcos) [inline doc-botonic/docs/faq.md:31]: Not necessary for this PR, but we'll need to add a deployment guide in the docs with 3 options:
* Self-hosted in browser (this is possible now)
* Self-hosted AWS (in progress)
* Deploy to Hubtype
We'll have to explain very well the limitations of each option.

File: doc-botonic/docs/welcome.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+> ## Chatbot vs Conversational App
+
+Before going any further, let us explain why Botonic is your best ally.
+
+Botonic is a React-based framework that allows you to build conversational experiences. 
+
+A chatbot is an intelligent artificial agent that implies an AI-centric, text-only interaction with your bot.
+
+**Botonic** helps you to create **Conversational App** to deliver a good user experience by using all the typical chatbot features like Guided flows, NLU, Multi-language, Media support (images, video, location...), Webviews, Human handoff or Webchat. Your Conversation app will learn fast, understand your need, know you and help for your transactions.
```
Comment (ericmarcos) [inline doc-botonic/docs/welcome.md:15]: I find this explanation confusing, let's discuss it in a call.

File: doc-botonic/docs/welcome.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
+---
+id: welcome
+title: Welcome to Botonic User Guide
+sidebar_label: Welcome to Botonic User Guide  
+---
+
+> ## Chatbot vs Conversational App
```
Comment (ericmarcos) [inline doc-botonic/docs/welcome.md:7]: I see the use of `>` in titles everywhere. This is not good for 2 reasons:
1) The UI that displays looks like a "quote" or an "alert", and that's the intended use. It makes the title look "less title".
2) It generates the HTML inside a `<blockquote>` tag, this will confuse Google crawlers and hurt our SEO.

If you want to change the general styles of titles, subtitles, etc, I'm sure there's a way to do it with custom CSS

File: doc-botonic/docs/plugins/21.botonic-plugins.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
+---
+id: botonic-plugins
+title: Create a Botonic Plugin
+---
+
```
Comment (ericmarcos) [inline doc-botonic/docs/plugins/21.botonic-plugins.md:5]: Creating a botonic plugin should be considered for advanced developers. I don't think this should be the 1st thing in the plugins section.
Proposed index:
1) What are Botonic plugins and why are they useful (we can keep this short, take inspiration from https://www.gatsbyjs.org/docs/plugins)
2) Botonic Plugin Library (a list of all the plugins and how they work, much like https://www.gatsbyjs.org/plugins/)
3) Authoring Plugins (how to create your own plugin)

File: doc-botonic/Dockerfile (hunk 1 above)
Comment (ericmarcos) [reply doc-botonic/Dockerfile:1]: Why do we need Docker here?

File: doc-botonic/website/sidebars.json [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+	"templates/childs",
+	"templates/template-dynamic-carousel",
+	"templates/template-human-handoff",
+	"templates/template-nlu"
+	],
+	
+	"Components": [
+	"components/components",
+	"components/co-text",
+	"components/co-buttons",
```
Comment (vanbasten17) [inline doc-botonic/website/sidebars.json:23]: would we better to have only `components/buttons` instead of `components/co-buttons`, as buttons already belong to this subpath url. If so, remember to modify all "co-" references

File: doc-botonic/docs/faq.md (hunk 2 above)
Comment (annehubtype) [reply doc-botonic/docs/faq.md:18]: - The menu problem is resolved.
- Detailed API component not added in this PR. I will add it for the next sprint.
- doc-botonic/docs/faq.md updated.

File: doc-botonic/docs/faq.md (hunk 3 above)
Comment (annehubtype) [reply doc-botonic/docs/faq.md:27]: FAQ updated.

File: doc-botonic/docs/faq.md (hunk 4 above)
Comment (annehubtype) [reply doc-botonic/docs/faq.md:35]: The idea was to add in the FAQs the questions asked among the community that are not in the doc for the moment. Must see if it makes sense then.

File: doc-botonic/docs/faq.md (hunk 5 above)
Comment (annehubtype) [reply doc-botonic/docs/faq.md:31]: I'll add it to the next sprint.

File: doc-botonic/website/sidebars.json (hunk 9 above)
Comment (annehubtype) [reply doc-botonic/website/sidebars.json]: Path renamed and updated.

File: doc-botonic/docs/plugins/21.botonic-plugins.md (hunk 8 above)
Comment (annehubtype) [reply doc-botonic/docs/plugins/21.botonic-plugins.md:5]: To do in next sprint then.

File: doc-botonic/docs/welcome.md (hunk 7 above)
Comment (annehubtype) [reply doc-botonic/docs/welcome.md]: Block removed.

**Comment by annehubtype** [comment]
Merging and adding to another branch.
